### PR TITLE
Add paging and archive/private repository filtering to dashboard list

### LINKED ISSUE #11268: feature: Hide archived repositories from Web UI
So I have been looking at archiving a lot of our old repositories. I see the option to archive which effectively makes the repo read-only and such. However, I see no option in the UI or the gitea ini to **hide** archived repositories from the Web UI. I have checked the forums and github to figure out if anything like this is available, and the only thing I have found is the suggestion to create a new org for archived repos.

However, this does not solve my original problem, because the repos still show up in the feed list. What I am trying to do is to have them as if they are no longer there. They are archived, I don't need to see them, but I don't quite need to delete them. I figure this could be a nice option to have, a checkbox either system-wide or user-specific to "hide" archived repositories from view. 

image

It would greatly help organization and cleanness.

### PR DESCRIPTION
**Pull request by zeripath**
Add paging to the dashboard repository list and additional filters for archived and private repositories.

Also fixes the tooltips and dropdowns from within the vue application.

Also updates the url as you change things so that on window reload you restore state.

repolist-other-filters
repolist-page-1
repolist-page-2

Fix #11268 
Closes #10674

### CODE REVIEW

File: models/repo_list.go [hunk 1]
```diff
@@ -205,14 +210,26 @@ func SearchRepositoryCondition(opts *SearchRepoOptions) builder.Cond {
...
-		accessCond := builder.Or(
-			//   A. Aren't in organisations  __OR__
-			builder.NotIn("owner_id", builder.Select("id").From("`user`").Where(builder.Eq{"type": UserTypeOrganization})),
-			//   B. Isn't a private or limited organisation.
-			builder.NotIn("owner_id", builder.Select("id").From("`user`").Where(builder.Or(builder.Eq{"visibility": structs.VisibleTypeLimited}, builder.Eq{"visibility": structs.VisibleTypePrivate}))))
-		cond = cond.And(accessCond)
+		// We should be able to see all non-private repositories that
+		// isn't in a private or limited organisation.
+		cond = cond.And(
+			builder.Eq{"is_private": false},
```
Comment (guillep2k) [inline models/repo_list.go:216]: We should cache this kind of flag in the repo itself at some future time. These queries are impossible.

**Review by guillep2k — APPROVED** [review]

File: models/repo_list.go (hunk 1 above)
Comment (zeripath) [reply models/repo_list.go:216]: Yeah they're too expensive and fiddly. 

It's so error prone. 

Really we just need your accessibility table...

**Comment by lunny** [comment]
Should we have an `all` filter?

**Comment by zeripath** [comment]
that's what the indeterminate state does.

**Comment by mrsdizzie** [comment]
Since this won't remember settings across page refresh I think having the archive option unchecked would be a better default option. If not, you'd have to remember to click it each time to filter archived repos and I bet most people would rather archived repos not shown by default if they can't set their own preferences. Would that seem reasonable?

**Comment by zeripath** [comment]
I didn't want to break existing behaviour - it would be a breaking change to make that happen. 

I guess to handle persistence we could just start adding to the window url query component?

**Comment by 6543** [comment]
have not looked deeper int this but can we hide the pagination element when there is only one page?

**Comment by zeripath** [comment]
@6543 it already only appears if there is more than one page.

**Comment by zeripath** [comment]
OK @mrsdizzie it now updates the search params as you select things and change them.

**Review by mrsdizzie — APPROVED** [review]
@zeripath That change was not totally what I was talking about but is nice : ) 

Right now every time you leave the dashboard page and come back, the option to include archived repos is set to 'on' even if you previously checked it to off. So for the case of the linked issue, this doesn't necessarily hide them as you have to manually re-hide them each time you visit the dashboard page. I think having it unchecked by default would be a desirable default even if it changes how it works now. I feel its only half breaking because it isn't going to stop something that currently works just by upgrading, it just moves the task of clicking the archive button to people who want to show and search archived repos which I imagine is a smaller subset of users (perhaps im wrong though).

just a small preference anyway, lgtm : )

**Review by lafriks — APPROVED** [review]

**Comment by zeripath** [comment]
@mrsdizzie oh I see!! I think we could do more about saving options in general. I'll have a think

**Comment by bkraul** [comment]
It would make more sense to make this an account option. So that the hiding of the archived projects could take place at a more global level. For example, if I have the option set to hide archived repos, and I go to explore an org that has several archived repos, I will not see them. Just a thought. But thanks for this feature!

**Comment by zeripath** [comment]
Why don't we open an issue, or better a pr to set it as unarchived as default so we can change it for 1.12...

**Comment by zeripath** [comment]
@bkraul @mrsdizzie see #11440

**Comment by bkraul** [comment]
@zeripath From what I can see it still applies only to the dashboard, and not to the explorer.

**Comment by zeripath** [comment]
OK sorry I missed that - that'll be have to be another PR. Please open an issue.

**Pull request merged**